pubtools/pulplib/_impl/client/client.py

```diff
@@ -203,6 +203,35 @@ def search_repository(self, criteria=None):
             Repository, "repositories", criteria=criteria, search_options=search_options
         )
 
+    def search_content_by_type(self, content_type, criteria=None):
```

Comment:
well, all this is about "user" comfort we you want to provide here. Of course use case when user wants to get both rpm and srpm in one method call makes sense. Due to pulp implementation details it's not possible to do such call in one request, therefore logic for getting more then one content type is simply do more then one request and merge results together. So you can implement search_content_by_type to do that for you or you can leave it up to user.
First approach provides more user comfort, however require more code to handle possible user inputs (see my conversation with Rohan above), second approach makes code easier and user had to do those steps "manually".
So I decided to rather leave it up to user as sequence of manual steps needed here is very simple compared to extra code needed for proper handling user input. 

Benefit vs effort needed for first option just seemed very unbalanced to me.

Reply:
I forgot Pulp only permits querying a single content type. I thought we had built something in to account for cases where _search_repo_units received more than one type ID but it looks like there's no handling for that and the user would get the Pulp error.

Reply:
> I forgot Pulp only permits querying a single content type, then thought we had built something in to account for cases where _search_repo_units received more than one type ID but it looks like there's no handling for that and the user would get the Pulp error.

No I don't think there's an error in that case.  It works, you can search for repo units over multiple type IDs at once.

There's an inconsistency in Pulp2's API here: when searching for units associated with a repo, you can search as many type IDs as you want in a single request.  When searching for units across multiple repos, you need to do a separate request per type ID.

The original approach in the pull request here has effectively passed along this inconsistency to the user of this library: if you want to search multiple type IDs in a single repo, you can do it by a single method call since Pulp works that way.  If you want to search multiple type IDs across all repos, you need one method call per type ID since Pulp works that way.

My view is that the library should instead be hiding this inconsistency rather than forcing library users to deal with it.

Reply:
Ah, I see. Thanks, @rohanpm.